core/extensions.rs

```diff
@@ -358,6 +373,11 @@ macro_rules! extension {
       }
 
       #[allow(dead_code)]
+      /// Initialize this extension for use with CommonJS
+      /// For modules, use init_ops_and_esm instead
+      ///
+      /// # Returns
+      /// an Extension object that can be used during instantiation of a JsRuntime
       pub fn init_js_only $( <  $( $param : $type + 'static ),* > )? () -> $crate::Extension
```

Comment:
I think we should leave this one undocumented, or even mark it as deprecated. We don't use it in CLI or anywhere I know of. And I don't know when you'd use it. It might be a legacy thing.

Reply:
```rust
      /// Legacy function for extension instantiation.
      /// Please use `init_ops_and_esm` or `init_ops` instead
      ///
      /// # Returns
      /// an Extension object that can be used during instantiation of a JsRuntime
      #[deprecated(since="0.216.0", note="please use `init_ops_and_esm` or `init_ops` instead")]
```

Added deprecation notice, and altered documentation to reflect the change